gnc/sub8_perception/src/sub8_pcl/buoy.cc

```diff
@@ -171,7 +218,10 @@ void Sub8BuoyDetector::determine_buoy_position(
       closest_buoy_index = buoy_index;
     }
   }
+
   center = buoy_centers[closest_buoy_index];
+
+  return true;
 }
 
 void Sub8BuoyDetector::image_callback(const sensor_msgs::ImageConstPtr &image_msg,
```

Comment:
I know this isnt from this PR but I'd suggest that you just update pointers to the image_msg and info_msg here and only the cv_bridge the messages inside your service callback function. That way we're only decoding images when they actually need to be processed.